lib/engine/part/base.rb

```diff
@@ -93,6 +93,10 @@ def blocks_lay?
         false
       end
 
+      def rect?
```

Comment:
does this need to be in the base? should it only be on nodes?

Reply:
Funny. I originally put it in Node and then decided to put it into base to match the other derived class overrides.